src/tokenizer/empty_tokenizer.rs

```diff
@@ -1,16 +1,17 @@
-use crate::tokenizer::{BoxTokenStream, Token, TokenStream, Tokenizer};
+use crate::tokenizer::{Token, TokenStream, Tokenizer};
 
 #[derive(Clone)]
 pub(crate) struct EmptyTokenizer;
 
 impl Tokenizer for EmptyTokenizer {
-    fn token_stream<'a>(&self, _text: &'a str) -> BoxTokenStream<'a> {
-        EmptyTokenStream::default().into()
+    type TokenStream<'a> = EmptyTokenStream;
+    fn token_stream<'a>(&self, _text: &'a str) -> EmptyTokenStream {
+        EmptyTokenStream::default()
```

Comment:
```suggestion
        EmptyTokenStream
```

Reply:
`EmptyTokenStream` isn't an empty struct, it contains a default `Token`